containers-contrib-readonly/src/Map/Contrib/Laws.hs

```diff
@@ -1,8 +1,51 @@
+{-# LANGUAGE TypeFamilies #-}
+
 module Map.Contrib.Laws
        ( singletonFromList
+       , nullEmpty
+       , emptyZeroSized
+       , singletonOneSized
+       , memberEmptyFalse
+       , newMemberExists
+       , newMemberYieldsValidValue
+       , lookupDefaultEmpty
+       , listToSingleton
+       , keysOfSingleton
+       , elemsOfSingleton
        ) where
 
-import Map (Key, fromList, singleton)
+import Map
+import Prelude hiding (lookup, null)
+
+nullEmpty :: Bool
+nullEmpty = null empty
+
+emptyZeroSized :: Bool
+emptyZeroSized = size empty == 0
```

Comment:
Yes you are completely right. I was trying to avoid `==>` since it makes us lose data, and we need to make sure to use small generators or QuickCheck quickly gives up.

It did not take much effort after all, I made the necessary changes and ping you as soon as my work is completed.